src/Maestro/Maestro.ContainerApp/Api/Controllers/ControllerConfiguration.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Maestro.AzureDevOps;
+using Maestro.DataProviders;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.GitHub.Authentication;
+using Microsoft.DotNet.Internal.Logging;
+using Microsoft.DotNet.Kusto;
+using Microsoft.DotNet.Services.Utility;
+using Microsoft.Extensions.Internal;
+
+namespace Maestro.ContainerApp.Api.Controllers;
+
+internal static class ControllerConfiguration
+{
+    public static void AddControllerConfigurations(this WebApplicationBuilder builder)
+    {
+        builder.Services.AddSingleton<IGitHubClientFactory, GitHubClientFactory>();
+        builder.Services.AddScoped<IRemoteFactory, DarcRemoteFactory>();
+        builder.Services.AddKustoClientProvider("Kusto");
+        builder.Services.AddGitHubTokenProvider();
+        builder.Services.AddSingleton<ISystemClock, SystemClock>();
+        builder.Services.AddSingleton<ExponentialRetry>();
```

Comment:
@dkurepa I wouldn't really make any Singletons, it's quite dangerous

Reply:
I copied things from how we're using them in the real code

Reply:
That's sad :D